### PR TITLE
feat(console): assign roles to user

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

- Assign roles to user
- Add the `UserRolesTranfer` component
- Extract common transfer styles into `transfer.module.scss`

### Todo 
- Share transfer component styles: LOG-5148

## Testing

image
image

### CODE REVIEW

File: packages/phrases/src/locales/en/translation/admin-console/user-details.ts [hunk 1]
```diff
@@ -46,6 +46,14 @@ const user_details = {
...
     delete_description: 'TBD', // UNTRANSLATED
     deleted: 'The role {{name}} has been successfully deleted from the user.', // UNTRANSLATED
+    assign_title: 'Assign roles to {{name}}',
+    assign_subtitle: 'Authorize {{name}} one or more roles',
+    assign_role_field: 'Assign roles',
+    role_search_placeholder: 'Search by role name',
+    added_text: '{{value, number}} added',
+    assigned_user_count: '{{value, number}} users',
+    confirm_assign: 'Assign roles',
+    role_assigned: 'Roles successfully assigned!',
```
Comment (gao-sun) [inline packages/phrases/src/locales/en/translation/admin-console/user-details.ts:56]:
```suggestion
    role_assigned: 'Successfully assigned role(s)',
```

File: packages/console/src/pages/UserDetails/UserRoles/index.tsx [hunk 2]
```diff
@@ -131,6 +132,17 @@ const UserRoles = () => {
...
+        <AssignRolesModal
+          user={user}
+          onClose={(success) => {
+            if (success) {
+              void mutate();
+            }
+            setIsAssignRolesModalOpen(false);
+          }}
+        />
+      )}
```
Comment (gao-sun) [inline packages/console/src/pages/UserDetails/UserRoles/index.tsx:145]: i think the modal logic with a state to control open/close is common, should we wrap it into a hook?

Comment (gao-sun) [reply packages/console/src/pages/UserDetails/UserRoles/index.tsx:145]: if so, no need to do it now

File: packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+              key={role.id}
+              role={role}
+              isSelected={isSelected}
+              onSelect={() => {
+                if (isSelected) {
+                  onChange(selectedRoles.filter(({ id }) => id !== role.id));
+
+                  return;
+                }
+                onChange([role, ...selectedRoles]);
```
Comment (gao-sun) [inline packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx:93]: nit: `onChange(isSelected ? ... : ...)`

File: packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  const isRoleSelected = (role: RoleResponse) =>
+    selectedRoles.findIndex(({ id }) => role.id === id) >= 0;
+
+  const handleSearchInput = (event: ChangeEvent<HTMLInputElement>) => {
+    clearTimer();
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    timerRef.current = setTimeout(() => {
+      setPageIndex(1);
+      setKeyword(event.target.value);
+    }, searchDelay);
```
Comment (gao-sun) [inline packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx:65]: this is a common pattern of debounce or something too, a hook?

File: packages/phrases/src/locales/en/translation/admin-console/user-details.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/phrases/src/locales/en/translation/admin-console/user-details.ts]: Updated.

File: packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx (hunk 3 above)
Comment (xiaoyijun) [reply packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx]: Updated.

File: packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/components/UserRolesTransfer/components/SourceRolesBox/index.tsx]: Good call, updated.

File: packages/console/src/pages/UserDetails/UserRoles/index.tsx (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/pages/UserDetails/UserRoles/index.tsx:145]: Add an issue to track this:
https://linear.app/silverhand/issue/LOG-5163/align-modal-open-logic

**Review by wangsijie — APPROVED** [review]

**Pull request merged**